src/mixer/playermanager.cpp

```diff
@@ -729,6 +734,13 @@ void PlayerManager::slotAnalyzeTrack(TrackPointer track) {
     }
 }
 
+void PlayerManager::slotSaveEjectedTrack(TrackPointer track) {
+    VERIFY_OR_DEBUG_ASSERT(track) {
+        return;
+    }
+    m_pLastEjectedTrack = track;
```

Comment:
Keeping more tracks loaded in-memory for an unforeseeable and unlimited amount of time will only fuel the various issues we already experience with delayed database updates and metadata synchronization. Uncontrolled accumulation of state gets you into trouble.

Reply:
The **trackId** can be stored without issues, right?

Reply:
Sure.

Reply:
Performance wise it is a good idea to use the pointer.

How is the situation with our cache? Is guaranteed that the last ejected track is kept in cache? 
Holding the pointer here enforces that, which is a good idea from that perspective.

The the issue with this is that it is hidden behind the scenes and it is hard to predict when the metadata is written to the file. We are already in that state, right? Is this relevant here?

Reply:
I see stale search results all the time, caused by these kind of issues. It's not only about metadata.

And it is bad practice to keep data alive for an unlimited of time. If you don't use the deck anymore the pointer is kept alive until Mixxx is finally closed.

Either store an ID are use a timer to reset the pointer periodically. I guess we have a similar caching issue in BaseTrackCache.

Reply:
An remember. These are fat QObjects with all the wiring that may cause all kinds of weird side-effects. This design is already unmaintainable. Don't make it worse.

Reply:
yeah we can switch this to an ID.

As to Daniel's question above, if the ID is invalid (because of something strange like the user deleted the file) the track load will simply fail

Reply:
Any hint for how to load a track by trackid?  I am a little lost in the library code

Reply:
`TrackCollecton::getTrackById()`

Reply:
this is fixed